orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/EvaluateDeploymentHealthTask.java

```diff
@@ -0,0 +1,105 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.monitoreddeploy;
+
+import com.netflix.spectator.api.Id;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.config.DeploymentMonitorServiceProvider;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthRequest;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.EvaluateHealthResponse;
+import com.netflix.spinnaker.orca.deploymentmonitor.models.StatusReason;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty(value = "monitored-deploy.enabled")
+public class EvaluateDeploymentHealthTask extends MonitoredDeployBaseTask {
+  @Autowired
+  EvaluateDeploymentHealthTask(
+      DeploymentMonitorServiceProvider deploymentMonitorServiceProvider, Registry registry) {
+    super(deploymentMonitorServiceProvider, registry);
+  }
+
+  @Override
+  public TaskResult executeInternal() {
+    EvaluateHealthRequest request = new EvaluateHealthRequest(stage);
+
+    log.info(
+        "Evaluating health of deployment {} at {}% with {}",
+        request.getNewServerGroup(), request.getCurrentProgress(), monitorDefinition);
+
+    EvaluateHealthResponse response = monitorDefinition.getService().evaluateHealth(request);
+    sanitizeAndLogResponse(response);
+
+    EvaluateHealthResponse.NextStepDirective directive = response.getNextStep().getDirective();
+
+    Id statusCounterId =
+        registry
+            .createId("deploymentMonitor.healthStatus")
+            .withTag("monitorId", monitorDefinition.getId())
+            .withTag("status", directive.toString());
+
+    registry.counter(statusCounterId).increment();
+
+    if (directive != EvaluateHealthResponse.NextStepDirective.WAIT) {
+      // Log timing
+      Id timerId =
+          registry
+              .createId("deploymentMonitor.timing")
+              .withTag("monitorId", monitorDefinition.getId())
+              .withTag("status", directive.toString());
+
+      double duration = Instant.now().toEpochMilli() - stage.getStartTime();
+      registry.gauge(timerId).set(duration);
```

Comment:
not sure I follow the use of a gauge here and not an actual timer?

Reply:
I think the timer in spectator is actually a gauge under the covers but i didn't want to/couldn't use `com.netflix.spectator.api.patterns.LongTaskTimer.get` because it stores state, hence the artisanal timer, I couldn't find a way to actually log a timer in `spectator`, did i miss sometihng?

Reply:
there are definitely `registry.timer(id).record(duration, unit)` methods

Reply:
oh, that's much better ,thank you!